test/test_enzyme.jl

```diff
@@ -1,13 +1,80 @@
+include("dependencies_for_runtests.jl")
+
 using Oceananigans
+using Oceananigans.TurbulenceClosures: with_tracers
+using Oceananigans.BoundaryConditions: fill_halo_regions!
+using Oceananigans.Fields: ConstantField
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: tracernames
 using Enzyme
 
 # Required presently
 Enzyme.API.runtimeActivity!(true)
-
-EnzymeRules.inactive_type(::Type{<:Oceananigans.Grids.AbstractGrid}) = true
+Enzyme.API.looseTypeAnalysis!(true)
```

Comment:
It would be cool to see if we can remove either of those flags (runtimeActivity/looseTypeAnalysis)

Reply:
Removed runtimeActivity and it worked, trying it without looseTypeAnalysis now